library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelListViewModel.java

```diff
@@ -214,6 +216,31 @@ public void onError(String errMsg, int errCode) {
         });
     }
 
+    /**
+     * removes the channel. Messages are permanently removed. //TODO I am not sure this method is needed. Client can call Channel.delete or Client.deleteChannel directly.
+     *
+     * @param channel  the channel needs to delete
+     * @param callback the result callback
+     */
+    public void deleteChannel(@NotNull Channel channel, @Nullable ResultCallback<Channel, String> callback) {
```

Comment:
I think a better approach is to remove this method from the view model and instead let the developer user `client.deleteChannel`. The view model then can register to `channel.deleted` events to update the list of channels